src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,148 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static final double CAPACITY_EXPAND_FACTOR = 1.5;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[CAPACITY];
+    }
+
+    public ArrayList(List<T> list) {
+        objects = new Object[CAPACITY];
+        if (list.size() <= CAPACITY) {
+            copyListToArrayListAndResize(list);
+            return;
+        }
+        do {
+            expandObjectsArray(objects);
+        } while (list.size() <= CAPACITY);
+        copyListToArrayListAndResize(list);
+    }
```

Comment:
the only function of constructors is an initialization of fields, you can't use such complex logic here, so remove this constructor

Reply:
Clear